pkg/catalog/catalog.go

```diff
@@ -891,22 +893,70 @@ func (c *Catalog) ListCommits(ctx context.Context, repository string, branch str
 		for _, parent := range v.Parents {
 			commit.Parents = append(commit.Parents, parent.String())
 		}
-		commits = append(commits, commit)
-		if len(commits) >= limit+1 {
+
+		if len(params.PathList) != 0 && len(v.Parents) == 1 {
+			// if we need to log only commits associated to the path, and also the current commit
+			// is not a merge commit, then we check if the current commit contains changes to the path
```

Comment:
Can you please edit this comment so that it is clearer? it explains the logic you implemented. 

To log only commits associated with a path, we check the diff between etc etc..

Reply:
I'm not sure I fully understood, but I tried to make it more clear, and also the explanation about the diff is in the 'pathInCommit' function.